plugins/maciVoting/contexts/CoordinatorContext.tsx

```diff
@@ -191,13 +179,64 @@ export const CoordinatorProvider = ({ children }: { children: ReactNode }) => {
     [getAuthorizationHeader, getPublicKey, publicClient]
   );
 
+  const checkMergeStatus = async (pollId: number) => {
+    const { address: pollAddress } = await getPoll({
+      maciAddress: PUBLIC_MACI_ADDRESS,
+      pollId,
+      signer,
+    });
+    const pollAbi = parseAbi(["function stateMerged() view returns (bool)"]);
+    const statemMerged = await publicClient.readContract({
+      address: pollAddress as `0x${string}`,
+      abi: pollAbi,
+      functionName: "stateMerged",
+    });
+    return statemMerged;
+  };
+
+  const finalizeProposal = useCallback(async (pollId: number) => {
+    setFinalizeStatus("merging");
+    const hasMerged = await checkMergeStatus(pollId);
+    if (!hasMerged) {
+      const mergeResult = await merge(pollId);
+      if (!mergeResult.success) {
+        return;
+      }
+    }
+    setFinalizeStatus("merged");
+
+    // TODO: get these values
+    const encryptedCoordinatorPrivateKey = "";
```

Comment:
This variable is divided in two:

1. The Coordinator Maci Keypair (it is supposed to be created outside the UI and setup in the https://github.com/privacy-scaling-explorations/maci-voting-plugin-aragon deployment

2. The Coordinator Maci Keypair is encrypted using the RSA from the coordinator service `/v1/proof/publicKey` (https://github.com/privacy-scaling-explorations/maci/blob/068da5648fe41320a9c391a6274efaf3aa73384a/apps/coordinator/ts/proof/proof.controller.ts#L96)